src/scroll-timeline-base.js

```diff
@@ -350,28 +358,30 @@ function getContainingBlock(element) {
       return findClosestAncestor(element, isBlockContainer);
 
     case 'absolute':
+       return findClosestAncestor(element, isAbsoluteElementContainer);
+
     case 'fixed':
-      return findClosestAncestor(element, isAbsoluteOrFixedElementContainer);
+      return findClosestAncestor(element, isFixedElementContainer);
   }
 }
 
 function getScrollParent(node) {
   if (!node)
     return undefined;
 
-  const containingBlock = getContainingBlock(node);
-  if (!containingBlock)
-    return document.scrollingElement;
-
-  const style = getComputedStyle(containingBlock);
-  switch(style['overflow-x']) {
-    case 'auto':
-    case 'scroll':
-    case 'hidden':
-      return containingBlock;
-
-    default:
-      return getScrollParent(containingBlock);
+  while (true) {
+    const containingBlock = getContainingBlock(node);
+    if (!containingBlock)
```

Comment:
nit: I think it's slightly cleaner to move this to the while condition, e.g.:

```js
while (node = getContainingBlock(node)) {
  const style = getComputedStyle(node);
  switch(style['overflow-x']) {
      case 'auto':
      case 'scroll':
      case 'hidden':
        return node;
  }
}
return document.scrollingElement;
```

Reply:
done.  Also added missing special case rule for the body element.